Library.rb

```diff
@@ -0,0 +1,54 @@
+
+
+class Library
+  attr_accessor :books, :orders, :readers, :authors
+
+  def initialize(dataFileName)
+    @books    = []
+    @orders   = []
+    @readers  = []
+    @authors  = []
+    loadData (dataFileName)
+  end
+
+
+  def get_best_reader
+    @orders.group_by { |orders| orders[:reader] }.values.max_by(&:size).first[:reader]
+  end
+
+  def get_bestseller
+    @orders.group_by { |orders| orders[:book] }.values.max_by(&:size).first[:book]
+  end
+
+  def get_popularity(size)
+    @orders.group_by { |orders| orders[:book] }.map{ |book, orders| [book, orders.size] }.max_by(size){ |book, count| count }
```

Comment:
1. can we move it to method @orders.group_by { |orders| orders[:book] } ?
2. check logic please